### DISCUSSION TITLE
Chemical equilibrium calculation times

### DISCUSSION THREAD

**Discussion by JMCarmona10** [reaktoro/reaktoro#378] [General]
I'm curious about Reaktoro's ability to determine the time required for multiple reactions in different phases of a chemical system to achieve a state of equilibrium. This question arises from my understanding that some chemical equilibria tend to be slower or faster than others. I'm wondering if there's a way for Reaktoro to differentiate between these various time scales.

For example, consider a chemical system composed of solid and aqueous phases. When I run the Reaktoro simulation, I'll obtain results for all the species I've considered, and even the solid phase will reach equilibrium. However, I'm unsure about how long it would take for all the phases to achieve equilibrium, whether it's a matter of seconds, years, or etc. This is a question I've been pondering while working with Reaktoro. Since i want to couple it with my transport model.  

(For now, I'm assuming all reactions are instantaneous but that's not logical).

**Comment by allanleal** [comment]
You need to introduce chemical kinetics in your simulations. Here is an example:
https://reaktoro.org/tutorials/kinetics/kinetics-minerals-using-palandri-kharaka-model.html
